src/tdastro/astro_utils/passbands.py

```diff
@@ -2,257 +2,246 @@
+        )
 
-        Files used are from:
-        http://svo2.cab.inta-csic.es/svo/theory/fps3/index.php?mode=browse&gname=LSST
+    def _load_preset(self, preset):
+        """Load a pre-defined set of passbands.
 
-        Note
-        ----
-        This is a temporary solution to be replaced when we implement Pooch throughout the project.
+        Parameters
+        ----------
+        preset : str
+            The name of the pre-defined set of passbands to load.
+        """
+        if preset == "LSST":
+            self.passbands = {
+                "u": Passband("LSST", "u"),
+                "g": Passband("LSST", "g"),
+                "r": Passband("LSST", "r"),
+                "i": Passband("LSST", "i"),
+                "z": Passband("LSST", "z"),
+                "y": Passband("LSST", "y"),
+            }
+        else:
+            raise ValueError(f"Unknown passband preset: {preset}")
+
+    def fluxes_to_bandfluxes(self, flux_density_matrix, wavelengths_angstrom) -> np.ndarray:
+        """Calculate bandfluxes for all passbands in the group.
 
         Parameters
         ----------
-        band_name : str
-            Name of the band for which the transmission table is to be downloaded.
-        output_file : str
-            Path to the file where the transmission table will be saved.
-        timeout : int
-            Time in seconds to wait for the file to be downloaded and created.
+        flux_density_matrix : np.ndarray
+            A 2D array of flux densities where rows are times and columns are wavelengths.
+        wavelengths_angstrom : np.ndarray
+            An array of wavelengths in Angstroms corresponding to the columns of flux_density_matrix.
 
         Returns
         -------
-        bool
-            True if the transmission table is successfully downloaded, False otherwise.
+        dict
+            A dictionary of bandfluxes with passband labels as keys.
         """
-        try:
-            socket.setdefaulttimeout(timeout)
+        bandfluxes = {}
+        for label, passband in self.passbands.items():
+            bandfluxes[label] = passband.fluxes_to_bandflux(flux_density_matrix, wavelengths_angstrom)
+        return bandfluxes
 
-            url = (
-                f"http://svo2.cab.inta-csic.es/svo/theory/fps3/getdata.php"
-                f"?format=ascii&id=LSST/LSST.{band_name}"
-            )
 
-            urllib.request.urlretrieve(url, output_file)
+class Passband:
+    """A passband contains information about its transmission curve and calculates its normalization.
 
-            if os.path.getsize(output_file) == 0:
-                logging.error(f"Transmission table for {band_name} is empty.")
-                return False
+    Attributes
+    ----------
+    label : str
+        The label of the passband.
+     survey : str
+        The survey to which the passband belongs.
+    full_name : str
+        The full name of the passband.
+    table_path : str
```

Comment:
I'm not sure `table_path` and `table_url` need to be attributes. You could just pass them along to the respective functions during `__init__`. It's not a problem to keep things as-is for now though.